lib/msal-core/src/AuthResponse.ts

```diff
@@ -11,12 +12,13 @@ export type AuthResponse = {
     uniqueId: string;
     tenantId: string;
     tokenType: string;
-    idToken: IdToken;
+    idToken: string;
```

Comment:
Seems like this is a breaking change, going from an object to a string?

Reply:
Yes, it is. This is a conscious one to keep the idToken object private:

- our internal naming of some claims in idToken can confuse users.
- cleaner interfacing, uniformity with accessToken format.

Reply:
Could this instead add a property (`idTokenString` or something) instead of changing the shape?

Reply:
We had that as "rawIdToken" part of idToken, but since we expose accessToken as a string it makes sense to do the same with idToken.

Reply:
okay so just to be clear... 
this used to be a raw id token and is still a raw id token right?

Reply:
nope, It used to be 'idToken: IdToken', which is a class in our library. What @pkanher617 is referring to is that that class also had a property, 'IdToken.rawIdToken' which is a "string".
The catch is that we never exported 'IdToken', hence it was just a js object.

Reply:
okay, my misunderstanding. I think I would agree that we should not introduce a breaking change. 
IdToken should stay an object, and we should deprecate it in major version.

Reply:
can also add rawIdToken separately

Reply:
@DarylThayil This is exactly the point I was making the other day before we decided on the interface.

- Though we have idToken as a JS object, we are not exporting its type.
- We need to either add 'IdToken' class as an exported type or change it to a rawToken.

Cons of exposing the internal non-standard IdToken class:

- The problem with exposing a non-standard  [IdToken](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/src/IdToken.ts) class is that we are choosing to part open the idToken for our internal code and having the developer see it.
- At the end of the day, all the dev needs is the idToken and the claims with it.
- This also gives the freedom to change/add to the internal IdToken class without affecting the developer.

This is the reason why I was advocating to change the current response to idToken: string instead of idToken: object. The object has been mysterious anyways unless someone opens up the code as we stand today.

Follow this up with good documentation, we should be able to push it.

Reply:
If you want to break this in a major version, I am fine with it. Let the PR stay and integrate it in 1.1.x!

Reply:
> We need to either add 'IdToken' class as an exported type or change it to a rawToken.

This is because since the `IdToken` class isn't exported, it doesn't get included in the exported types for the library, and thus doesn't show up in intellisense?

I would personally advocate for exporting the class (and having teh types work), instead of introducing a breaking change and creating work for our users.

Reply:
@jasonnutter From a code perspective, I get your stand. My concern is more about exposing the non-standard idToken class type that MSAL JS introduced. No MSAL exposes an IdToken class that picks only certain claims in the class structure and renames them internally.

It should not have been returned to the user in the firs place IMO.

Reply:
Made the changes non-breaking. Please review.